src/js/ripple/amount.js

```diff
@@ -342,8 +343,12 @@ Amount.prototype._check_limits = function() {
   if (this._value.isNaN() || this._value.isZero()) {
     return this;
   }
-  if (!this._is_native) {
-    var absval = this._value.absoluteValue();
+  var absval = this._value.absoluteValue();
+  if (this._is_native) {
+    if (absval.greaterThan(Amount.bi_xrp_max)) {
+      throw new Error('Exceeding max value of ' + Amount.bi_xrp_max.toString());
+    }
+  } else {
     if (absval.lessThan((new BigNumber(Amount.min_value)).absoluteValue())) {
       throw new Error('Exceeding min value of ' + Amount.min_value);
```

Comment:
It might be worth making some `bi_min_value` and `bi_max_value` BigNumber constants, given we do these checks a lot ?